### ISSUE TITLE
Synchronization problem of network editor while adding interaction

### ISSUE THREAD

**Issue by metincansiper** [PathwayCommons/factoid#313]
## Environment info

- Browser version (client) : Google Chrome Version 64.0.3282.140
- Node.js version (server) : 8.11.2

## Current (buggy) behaviour

The following error is thrown from network editor while working on the same document on 2 synchronized network editor.

Unhandled rejection TypeError: Cannot read property '_private' of undefined
    at Animation.hook (http://localhost:3000/deps.js:40488:28)
    at Animation.play (http://localhost:3000/deps.js:40519:10)
    at applyEditAnimationAsOverlay (http://localhost:3000/bundle.js:10730:21)
    at applyEditAnimation (http://localhost:3000/bundle.js:10741:12)
    at http://localhost:3000/bundle.js:10945:7
    at onDoc (http://localhost:3000/bundle.js:10693:5)
    at Interaction.onDocRemRetypePpt (http://localhost:3000/bundle.js:10936:5)
    at EventEmitter.emit (http://localhost:3000/deps.js:27776:35)
    at emit (http://localhost:3000/bundle.js:53:22)
    at Interaction.emit (http://localhost:3000/bundle.js:56:5)
    at Interaction.<anonymous> (http://localhost:3000/bundle.js:2766:13)
    at EventEmitter.emit (http://localhost:3000/deps.js:27776:35)
    at http://localhost:3000/bundle.js:3720:29
    at Array.forEach (<anonymous>)
    at http://localhost:3000/bundle.js:3709:23
From previous event:
    at Syncher.<anonymous> (http://localhost:3000/bundle.js:3708:12)
    at EventEmitter.emit (http://localhost:3000/deps.js:27775:35)
    at emit (http://localhost:3000/bundle.js:53:22)
    at Syncher.emit (http://localhost:3000/bundle.js:56:5)
    at emit (http://localhost:3000/bundle.js:1511:17)
    at postValidateHash (http://localhost:3000/bundle.js:1538:11)
    at checkHash (http://localhost:3000/bundle.js:1532:13)
    at Socket.<anonymous> (http://localhost:3000/bundle.js:1547:9)
    at Socket.Emitter.emit (http://localhost:3000/deps.js:83112:20)
    at Socket.onevent (http://localhost:3000/deps.js:82824:10)
    at Socket.onpacket (http://localhost:3000/deps.js:82782:12)
    at Manager.<anonymous> (http://localhost:3000/deps.js:83199:15)
    at Manager.Emitter.emit (http://localhost:3000/deps.js:83112:20)
    at Manager.ondecoded (http://localhost:3000/deps.js:81868:8)
    at Decoder.<anonymous> (http://localhost:3000/deps.js:83199:15)
    at Decoder.Emitter.emit (http://localhost:3000/deps.js:115927:20)
    at Decoder.add (http://localhost:3000/deps.js:68931:12)
    at Manager.ondata (http://localhost:3000/deps.js:81858:16)
    at Socket.<anonymous> (http://localhost:3000/deps.js:83199:15)

## Desired behaviour

Error should not be thrown and the interaction must be visible on both tabs.

## Minimum steps to reproduce

Open the same document in 2 different network editor tabs. Add an interaction on tab-1. Check the tab-2. Seems like no problem on tab-1 but the interaction will not be added on tab-2. The error above will be thrown and the following result will be obtained:

<img width="96" alt="screen shot 2018-06-11 at 4 40 01 pm" src="https://user-images.githubusercontent.com/6534865/41262565-763f4292-6d96-11e8-9465-6eb14931be4c.png">

I am not totally sure for now but I think that interactions participant list is empty on tab-2 and that would be the cause of the problem.

#307 is also caused by that interactions participant list is empty on the other tab when an interaction is added from network editor.

**Comment by d2fong** [comment]
I followed the prescribed steps and could not reproduce this issue.

**Comment by metincansiper** [comment]
@d2fong would you share your rethinkdb version as well?

**Comment by d2fong** [comment]
rethinkdb 2.3.6 (CLANG 8.0.0 (clang-800.0.42.1))

**Comment by metincansiper** [comment]
That is very interesting. 

- My chrome version was earlier than yours I updated it to version 67.0.3396.87 since I could not find exactly the same version but it must be fine I think.
- My npm version was already the same.
- My node version was slightly newer I downgraded it to the exactly same version.
- About rethinkdb my version is 2.3.6 (CLANG 8.1.0 (clang-802.0.42)). I cannot downgrade it to CLANG 8.0.0. Actually I could not understand what does clang stands for and whether it makes any difference.

As a result I am still having the same error.
